### DISCUSSION TITLE
Does this work in Web exports to bind the GDExtension API?

### DISCUSSION THREAD

**Discussion by Splizard** [godotjs/GodotJS#4] [Q&A]
Looking for a way to bridge into Godot's GDextension API via JS in the browser, is this project aiming to support this?

**Comment by ialex32x** [comment]
For now, it's not supported yet. But it seems possible to use quickjs as the javascript runtime which could run in the browser. I'll try to implement this bridging layer later, but may not very soon XD. (completely implement another bridging layer for quickjs, or add a wrapper layer which makes quickjs usable in the same manner as v8 and reuse the current bridge-v8)

And in this way, the scripts run on quickjs, rather than directly on the VM used by the browser, I'm not sure whether the performance can meet the production-level requirements or not.

**Reply by Splizard** [reply]
Oh, is there a reason you wouldn't use the browser's builtin Javascript for Web exports?

**Reply by ialex32x** [reply]
Oh, it's not a decision, it's only a preliminary thought, the method you mentioned is also great if it's feasible, just need to investigate it futher (I lack experience in this area 😄)

**Reply by ialex32x** [reply]
After further investigation, I think it's a good idea to bridge the runtime with the browser VM. It'll be implemented in a future version.

**Reply by Splizard** [reply]
@ialex32x let me know if there is anything I can do to help, I've reached out to the Godot web team (as well as GDExtension) and they're open to accepting contributions into the engine to support this. My understanding is that due to emscripten limitations, this cannot be done in a side module, so it likely needs to be part of the engine.

**Reply by ialex32x** [reply]
It works now (but very basically), scripts are directly run on the browser javascript VM.

**Reply by Splizard** [reply]
Ah nice! Any plans to support GDExtension? I see the project is currently built as a module, which introduces more friction for users to adopt versus including a .dll/.wasm/.so into their project.

**Reply by ialex32x** [reply]
I'll find a way to do so if there is no costs too much, but I haven't found an equivalent way to do `AddJSLibraries/AddJSPre`, it happens at the engine build time.

**Reply by Splizard** [reply]
@ialex32x graphics.gd now uses a GDExtension JS bridging layer for web exports (currently maintained in my fork of https://github.com/splizard/godot), this could help if you decide to investigate GDExtension in this way. I'm happy to help explain how this works to the GodotJS maintainers.
